src/swarm/neo/util/AcquiredResources.d

```diff
@@ -66,6 +66,26 @@ public struct AcquiredArraysOf ( T )
 
     /***************************************************************************
 
+        Gets a pointer to a new array, acquired from the shared resources pool.
+
+        Important note about array casting: it has been observed that code such
+        as the following can lead to segfaults
+
+        ---
+            AcquiredArraysOf!(void) arrays;
+            arrays.initialise(buffer_pool); // buffer_pool is assumed to exist
+            auto void_array = arrays.acquire();
+
+            struct S { int i; hash_t h; }
+            auto s_array = cast(S[]*)void_array;
+            s_array.length = 23;
+        ---
+
+        The exact reason for this is not known, but it appears to lead to
+        corruption of internal GC data (possibly type metadata associated with
+        the array's pointer). Setting the length of the array *before* casting
+        is safe.
+
```

Comment:
Suggest adding a positive example, such as
```d
(*void_array).length = 23 * S.sizeof; // First set the buffer length...
auto s_array = cast(S[])*void_array;   // ... then cast the array.
```
Maybe even change the order: First say how one should do it, then mention a way that is known to fail.

Reply:
Yep, good idea.

Reply:
Updated.